### PR TITLE
feature: external drag and drop in Compose Multiplatform for desktop

### PR DESCRIPTION
**Pull request by zamulla**

### CODE REVIEW

File: topics/compose/compose-drag-drop.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        override fun onStarted(event: DragAndDropEvent) {
+            showTargetBorder = true
+        }
+
+        override fun onEnded(event: DragAndDropEvent) {
+            showTargetBorder = false
+        }
+
+        override fun onDrop(event: DragAndDropEvent): Boolean {
+            println("Action at the target: ${event.action}")
```
[the remark this answers is not in the document]
Comment (zamulla) [reply topics/compose/compose-drag-drop.md:113]: There is no need, but it felt nice to immediately see in the log that it works as advertised, without the need to set up debugging output by myself. If this was more than a couple of lines, I'd probably find it cumbersome, but as it is, no big deal.

File: topics/compose/compose-drag-drop.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+[//]: # (title: Drag and drop operations)
+
+<title label="EAP" annotations="Desktop">Drag and drop operations</title>
+
+To allow users to drag files and objects in or out of your Compose Multiplatform app,
+you can use the `dragAndDropSource` and `dragAndDropTarget` modifiers to specify potential sources or destinations for drag operations.
```
[the remark this answers is not in the document]
Comment (zamulla) [reply topics/compose/compose-drag-drop.md]: It's hard not to focus on drag and drop when that is the title of the page and the name of the API :)

I rewrote the paragraph a little more towards the abstract. "Sharing" intuitively is more like something that is happening behind the scenes (at least in the Apple ecosystem experience). I also tried "transfer" for size, but that made it sound network-adjacent.

File: topics/compose/compose-drag-drop.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
+[//]: # (title: Drag and drop operations)
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:1]: Here and below, "drag and drop" should be hyphenated when used as an adjective, drag-and-drop operations, drag-and-drop session, target, and so on

File: topics/compose/compose-drag-drop.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+Example of a `Box()` composable that allows users to drag a string from it:
+
+```kotlin
+val exportedText = "Hello, drag and drop!"
+
+Box(Modifier
+    .dragAndDropSource(
+        // Creates a visual representation for the data being dragged
+        // (white rectangle with the exportedText string centered on it).
+        drawDragDecoration = {
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:31]: Indentation inside drawDragDecoration is 3 spaces, not 4

File: topics/compose/compose-drag-drop.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+3. Describe the data that is supposed to be dragged to the target with a `DragAndDropTransferable()` call.
+
+Example of a `Box()` composable that allows users to drag a string from it:
+
+```kotlin
+val exportedText = "Hello, drag and drop!"
+
+Box(Modifier
+    .dragAndDropSource(
+        // Creates a visual representation for the data being dragged
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:29]:
```suggestion
        // Creates a visual representation of the data being dragged
```

File: topics/compose/compose-drag-drop.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                    // system output with onTransferCompleted().    
+                    DragAndDropTransferData(
+                        transferable = DragAndDropTransferable(
+                            StringSelection(exportedText)
+                        ),
+
+                        // List of actions supported by this drag source. A type of action
+                        // is passed to the drop target together with data.
+                        // The target can use this to reject an inappropriate drop operation
+                        // or to interpret user's expectations.
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:68]: I think it works without possessive too, up to you

File: topics/compose/compose-drag-drop.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+```
+{initial-collapse-state="collapsed"  collapsed-title="Box(Modifier.dragAndDropSource"}
+
+## Creating a drop target
+
+To prepare a composable to be a drag and drop target:
+
+1. Describe the conditions for the composable to be the target of a drop in the `shouldStartDragAndDrop` lambda.
+2. Create (and `remember`) the `DragAndDropTarget` object that will contain your overrides for drag event handlers.
+3. Write the necessary overrides: for example, `onDrop` for parsing the received data, or `onEntered` when a draggable
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:96]:
```suggestion
3. Write the necessary overrides: for example, `onDrop` for parsing the received data or `onEntered` when a draggable
```

File: topics/compose/compose-drag-drop.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    )
+) {
+    Text(targetText, Modifier.align(Alignment.Center))
+}
+```
+{initial-collapse-state="collapsed"  collapsed-title="val dragAndDropTarget = remember"}
+
+## What's next
+
+For more implementation details and common use cases, see the [Drag and drop](https://developer.android.com/develop/ui/compose/touch-input/user-interactions/drag-and-drop) article about the corresponding modifiers
```
Comment (danil-pavlov) [inline topics/compose/compose-drag-drop.md:166]:
```suggestion
For more details on implementation and common use cases, see the [Drag and drop](https://developer.android.com/develop/ui/compose/touch-input/user-interactions/drag-and-drop) article about the corresponding modifiers
```

**Review by danil-pavlov — CHANGES_REQUESTED** [review]

File: topics/compose/compose-drag-drop.md (hunk 3 above)
Comment (danil-pavlov) [reply topics/compose/compose-drag-drop.md:1]: The top title should also be fixed, otherwise LGMT

**Review by danil-pavlov — APPROVED** [review]

**Pull request merged**